### PR TITLE
fix: missing item in docs table of contents

### PR DESCRIPTION
**Pull request by Chriscbr**
Small error. Also fixing a broken link

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/github.md [hunk 1]
```diff
@@ -10,4 +10,4 @@ TODO
 
 See the `GitHub`, `GithubWorkflow`, and `Job` types in the [API reference](./api/API.md) for currently available APIs.
```
Comment (kaizen3031593) [inline docs/github.md:11]: Should we change the links to the API reference to the new docs page? There's a link to the API reference in the `docs/README.md` file too. Both seem outdated.

Comment (Chriscbr) [reply docs/github.md:11]: good catch! Updating

Comment (Chriscbr) [reply docs/github.md:11]: Ok I don't think there are any direct references to projen.github.io anymore (although I think it still redirects if you visit that URL which is fine) - not sure if I'm misunderstanding 😅

Comment (kaizen3031593) [reply docs/github.md:11]: !! I didn't even know the url was changed! projen.io sounds way snazzier. I meant line 26 in the docs/README.md file-- it points to `api/API.md` and I think it should point to `https://projen.io/api/API.html`

Comment (Chriscbr) [reply docs/github.md:11]: Oh gotcha. That's a relative link, which should automatically resolve in most places - if I click on the link at the bottom of https://projen.io/ or at the bottom of https://github.com/projen/projen/blob/main/docs/api/API.md it takes me to the correct page. If it doesn't work somewhere I'm ok with hardcoding it though

Comment (kaizen3031593) [reply docs/github.md:11]: ah. i was looking at it on github and it links to the github api.md page that is too large to render. up to you if you want to hard-code it or leave it as is

**Review by kaizen3031593 — APPROVED** [review]

**Pull request merged**